middleware/hosts/hosts.go

```diff
@@ -49,17 +50,21 @@ func (h Hosts) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 		answers = h.ptr(qname, names)
 	case dns.TypeA:
 		ips := h.LookupStaticHostV4(qname)
+		otherRecordCount = len(h.LookupStaticHostV6(qname))
 		answers = a(qname, ips)
 	case dns.TypeAAAA:
 		ips := h.LookupStaticHostV6(qname)
+		otherRecordCount = len(h.LookupStaticHostV4(qname))
 		answers = aaaa(qname, ips)
 	}
 
 	if len(answers) == 0 {
 		if h.Fallthrough {
 			return middleware.NextOrFailure(h.Name(), h.Next, ctx, w, r)
 		}
-		return dns.RcodeRefused, nil
+		if otherRecordCount == 0 {
+			return dns.RcodeRefused, nil
```

Comment:
I realize the response code hasn't changed, but why `refused` instead of `nxdomain` when no records exist?